### PR TITLE
[Do not merge] IgxTabs/IgxBottonNav routing support POC

### PR DESCRIPTION
**Pull request by rmkrmk**
#4297

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 1]
```diff
@@ -160,7 +167,7 @@ export class IgxTabsGroupComponent implements IgxTabsGroupBase, AfterContentInit
      *```
      * @param focusDelay A number representing the expected delay.
      */
-    public select(focusDelay = 200): void {
+    public select(focusDelay = 200, navigateToRoute: boolean = true): void {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: No need to mark this as `Boolean`, as assigning it to `true` by default makes it a Boolean type.

File: projects/igniteui-angular/src/lib/tabs/tabs.component.ts [hunk 2]
```diff
@@ -282,13 +284,25 @@ export class IgxTabsComponent implements IgxTabsBase, AfterViewInit, OnDestroy {
         }
     }
 
-    private setSelectedGroup(): void {
+    private setSelectedGroup(navigateToRoute: boolean = true): void {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/tabs/tabs.component.ts]: Same comment as before.

File: projects/igniteui-angular/src/lib/tabbar/tabbar.component.ts [hunk 3]
```diff
@@ -357,11 +413,20 @@ export class IgxTabPanelComponent implements AfterContentInit, AfterViewChecked
...
+     *@hidden
+     */
+    public selectAndUpdateRoute(updateRoute: boolean) {
         if (this.disabled || this._tabBar.selectedIndex === this.index) {
             return;
         }
-
         this.isSelected = true;
+        if (updateRoute && this.routerLinkDirective) {
+            this.routerLinkDirective.onClick();
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/tabbar/tabbar.component.ts]: Better do `this.router.navigate()`.

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts (hunk 1 above)
Comment (rmkrmk) [reply projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: Addressed.

File: projects/igniteui-angular/src/lib/tabs/tabs.component.ts (hunk 2 above)
Comment (rmkrmk) [reply projects/igniteui-angular/src/lib/tabs/tabs.component.ts]: Addressed.

**Comment by rmkrmk** [comment]
Closing this PR because this approach will not be used.

**Pull request closed without merging**